iroh-rpc-client/src/store.rs

```diff
@@ -66,22 +68,38 @@ impl StoreClient {
     }
 
     #[tracing::instrument(skip(self))]
-    pub async fn check(&self) -> StatusRow {
-        let status: ServiceStatus = self
-            .version()
-            .await
-            .map(|_| ServiceStatus::Serving)
-            .unwrap_or_else(|_e| ServiceStatus::Unknown);
-        StatusRow {
+    pub async fn check(&self) -> ServiceStatus {
+        let (status, version) = match self.version().await {
+            Ok(version) => (StatusType::Serving, version),
+            Err(_) => (StatusType::Down, String::new()),
+        };
+        ServiceStatus {
             name: "store",
-            number: 3,
+            number: 1,
             status,
+            version,
         }
     }
 
     #[tracing::instrument(skip(self))]
-    pub async fn watch(&self) -> impl Stream<Item = StatusRow> {
-        // todo
-        futures::stream::pending()
+    pub async fn watch(&self) -> impl Stream<Item = ServiceStatus> {
```

Comment:
these seem to be the same, maybe you could extract this into a generic function that is called for these?

Reply:
attempted to do this and ran into some limitations. brought it up to @rklaehn and he is brainstorming.

Reply:
ok 👌

Reply:
issue filed: https://github.com/n0-computer/iroh/issues/614